pix2tex/dataset/latex2png.py

```diff
@@ -53,51 +58,64 @@ def convert_file(self, infile, workdir, return_bytes=False):
 
         try:
             # Generate the PDF file
-            cmd = 'xelatex -halt-on-error -output-directory %s %s' % (workdir, infile)
+            #  not stop on error line, but return error line index,index start from 1
+            cmd = 'xelatex -interaction nonstopmode -file-line-error -output-directory %s %s' % (
+                workdir, infile)
 
             p = subprocess.Popen(
-                cmd,
-                shell=True,
+                shlex.split(cmd),
                 stdin=subprocess.PIPE,
                 stdout=subprocess.PIPE,
                 stderr=subprocess.PIPE,
+                universal_newlines=True
             )
             sout, serr = p.communicate()
-            # Something bad happened, abort
-            if p.returncode != 0:
-                raise Exception('latex error', serr, sout)
-
+            # extract error line from sout
+            error_index, _ = extract(text=sout, expression="%s:(\d+)" % infile)
+            # extract success rendered equation
+            if error_index != []:
+                # offset index start from 0, same as self.math
+                error_index = [int(_)-self.prefix_line-1 for _ in error_index]
             # Convert the PDF file to PNG's
             pdffile = infile.replace('.tex', '.pdf')
+            result, _ = extract(
+                text=sout, expression="Output written on %s \((.*)? pages\)" % pdffile)
+            if int(result[0]) != len(self.math):
+                raise Exception('xelatex rendering error, generated %d formula\'s page, but the total number of formulas is %d.' % (
+                    int(result[0]), len(self.math)))
             pngfile = os.path.join(workdir, infile.replace('.tex', '.png'))
 
-            cmd = 'magick convert -density %i -colorspace gray %s -quality 90 %s' % (
+            cmd = 'convert -density %i -colorspace gray %s -quality 90 %s' % (
```

Comment:
This doesn't work for me now. I think it's platform specific. 
I've fixed it locally. Should run for us both. I'll push some changes in a sec.

Reply:
OK, I haven't figured out how to fix it...Looking forward to your revision